webapp/src/routes/app/+page.svelte

```diff
@@ -1,53 +1,221 @@
+    let zoom = 10;
+
+    let searchQuery = '';
+    let results = [];
+    let dropdownOpen = false;
+    let selectedListingId: string | null = null; // to track the selected listing for highlighting
+
+    let showBackToTop = false;
+    let listingsContainer: HTMLDivElement;
+
+    const searchLocation = async () => {
+        const url = `https://api.geoapify.com/v1/geocode/search?text=${encodeURIComponent(searchQuery)}&apiKey=${apiKey}`;
+
+        try {
+            const response = await fetch(url);
+            const data = await response.json();
+            results = data.features || [];
+            dropdownOpen = true;
+        } catch (error) {
+            console.error('Error fetching data:', error);
+        }
+    };
+
+    const handleSelect = (location: {
+        properties: { formatted: string };
+        geometry: { coordinates: number[] };
+    }) => {
+        searchQuery = location.properties.formatted;
+        mapCenter = [location.geometry.coordinates[0], location.geometry.coordinates[1]];
+        zoom = 13;
+        results = [];
+        dropdownOpen = false;
+    };
+
+    $: if (searchQuery && searchQuery.length > 3) {
+        searchLocation();
+    }
+
+    const handleClickOutside = (event: Event) => {
+        if (
+            !(event.target as HTMLElement).closest('.dropdown') &&
+            !(event.target as HTMLElement).closest('.search-input')
+        ) {
+            dropdownOpen = false;
+        }
+    };
+
+    onMount(() => {
+        document.addEventListener('click', handleClickOutside);
+        listingsContainer.addEventListener('scroll', handleScroll);
+        return () => {
+            document.removeEventListener('click', handleClickOutside);
+            listingsContainer.removeEventListener('scroll', handleScroll);
+        };
+    });
+
+    onDestroy(() => {
+        document.removeEventListener('click', handleClickOutside);
+        listingsContainer.removeEventListener('scroll', handleScroll);
+    });
+
+    const handleMarkerClick = (listingId: string) => {
+        selectedListingId = listingId;
+        document
+            .getElementById(`listing-${listingId}`)
+            ?.scrollIntoView({ behavior: 'smooth', block: 'center' });
+    };
+
+    const scrollToTop = () => {
+        if (listingsContainer) {
+            listingsContainer.scrollTo({ top: 0, behavior: 'smooth' });
+        }
+    };
+
+    const handleScroll = () => {
+        showBackToTop = listingsContainer.scrollTop > 0;
+    };
 </script>
 
 <Navbar />
 
-<body>
-    <div class="container">
-        <div class="listings">
-            <h2>Listings go here!</h2>
-        </div>
-        <div class="map-view">
-            <MapLibre
-                center={[-97.1, 49.9]}
-                zoom={10}
-                class="map"
-                standardControls
-                style="https://basemaps.cartocdn.com/gl/positron-gl-style/style.json"
-            />
-        </div>
+<div class="container">
```

Comment:
```suggestion
<div class="container" role="document" on:click={handleClickOutside}>
```

How about do this and you can get away from defining all the binding onMount?

Reply:
The conventional way of handling lifecycle hooks in Svelte is by using onMount, but I can change that if you want.